core/livepeernode.go

```diff
@@ -78,26 +88,36 @@ func NewLivepeerNode(e eth.LivepeerEthClient, vn net.VideoNetwork, nodeId NodeID
 		return nil, ErrLivepeerNode
 	}
 
-	return &LivepeerNode{VideoCache: NewBasicVideoCache(vn), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd, PeerConns: make([]PeerConn, 0)}, nil
+	return &LivepeerNode{VideoCache: NewBasicVideoCache(vn), VideoNetwork: vn, Identity: nodeId, Addrs: addrs, Eth: e, WorkDir: wd}, nil
 }
 
 //Start sets up the Livepeer protocol and connects the node to the network
-func (n *LivepeerNode) Start(ctx context.Context, bootID, bootAddr string) error {
+func (n *LivepeerNode) Start(ctx context.Context, bootIDs, bootAddrs []string) error {
+	if len(bootIDs) != len(bootAddrs) {
+		return errors.New("BootIDs and BootAddrs do not match.")
+	}
 	//Set up protocol (to handle incoming streams)
 	if err := n.VideoNetwork.SetupProtocol(); err != nil {
 		glog.Errorf("Error setting up protocol: %v", err)
 		return err
 	}
 
-	//Connect to bootstrap node.  This currently also kicks off a bootstrap process, which periodically checks for new peers and connect to them.
-	if bootID != "" && bootAddr != "" {
-		glog.Infof("Connecting to %v %v", bootID, bootAddr)
-		if err := n.VideoNetwork.Connect(bootID, []string{bootAddr}); err != nil {
-			glog.Errorf("Cannot connect to node: %v", err)
-		} else {
-			n.PeerConns = append(n.PeerConns, PeerConn{NodeID: bootID, NodeAddr: bootAddr})
+	//Connect to bootstrap node.
+	//TODO: Kick off a bootstrap process, which periodically checks for new peers and connect to them.
+	errCount := 0
+	if len(bootIDs) > 0 && len(bootAddrs) > 0 {
+		for i, bootID := range bootIDs {
+			bootAddr := bootAddrs[i]
+			glog.V(common.DEBUG).Infof("Connecting to %v %v", bootID, bootAddr)
+			if err := n.VideoNetwork.Connect(bootID, []string{bootAddr}); err != nil {
+				glog.Errorf("Cannot connect to node: %v", err)
+				errCount++
+			}
 		}
 	}
+	if errCount == len(bootIDs) {
+		glog.Errorf("Current node has no neighbors :(")
```

Comment:
Just curious: do we log the error here instead of returning the error with the expectation that the logic for kicking off a process to periodically check for new peers and connect to them will be added in this function at some point in the future in which case we wouldn't want to return an error because the node might still form connections later on during its periodic checks allowing it to actually function?

Reply:
I'm indifferent either way since it's in the boot-up sequence.